jupyter/sum-to-zero/sum_to_zero_evaluation.qmd

```diff
@@ -0,0 +1,1582 @@
+sim_df['raw_prev'] = sim_df['pos_tests'] / sim_df['tests']
+(
+    p9.ggplot(sim_df)
+    + p9.geom_density(p9.aes(x='raw_prev'), color='darkblue', fill='blue', alpha=0.3)
+    + p9.geom_density(p9.aes(x='p_sample'), color='darkorange', fill='pink', alpha=0.3)
+    + p9.labs(
+        x='raw prevalence',
+        y='',
+        title='Observed (blue) and underlying true prevalence (pink)\nlarge dataset'
+    )
+    + p9.theme_minimal()
+)
+```
+
+### Model Fitting
+
+Assemble the data dictionary of all input data for the model which solves the inverse problem -
+i.e., estimates regression coefficients given the observed data.
+We use the generated data as the inputs.
+Because the output files are real-valued outputs, regardless of variable element type,
+model data variables of type int need to be cast to int.
+Here all the observed data is count and categorial data.
+
+```{python}
+data_fixed = {'N':sim_data.pos_tests.shape[1], 
+              'N_age':gen_data_dict['N_age'], 
+              'N_eth':gen_data_dict['N_eth'],
+              'N_edu':gen_data_dict['N_edu'],
+              'sens': gen_data_dict['sens'],
+              'spec': gen_data_dict['spec'],
+              'intercept_prior_mean': gen_data_dict['baseline'],
+              'intercept_prior_scale': 2.5}
+
+data_small = data_fixed | {'pos_tests':sim_data.pos_tests[0].astype(int),
+             'tests':sim_data.tests[0].astype(int),
+             'sex':sim_data.sex[0].astype(int),
+             'age':sim_data.age[0].astype(int), 
+             'eth':sim_data.eth[0].astype(int),
+             'edu':sim_data.edu[0].astype(int)}
+
+data_large = data_fixed | {'pos_tests':sim_data_lg.pos_tests[0].astype(int),
+             'tests':sim_data_lg.tests[0].astype(int),
+             'sex':sim_data_lg.sex[0].astype(int),
+             'age':sim_data_lg.age[0].astype(int), 
+             'eth':sim_data_lg.eth[0].astype(int),
+             'edu':sim_data_lg.edu[0].astype(int)}
+
+data_tiny = data_fixed | {'pos_tests':sim_data_tiny.pos_tests[0].astype(int),
+             'tests':sim_data_tiny.tests[0].astype(int),
+             'sex':sim_data_tiny.sex[0].astype(int),
+             'age':sim_data_tiny.age[0].astype(int), 
+             'eth':sim_data_tiny.eth[0].astype(int),
+             'edu':sim_data_tiny.edu[0].astype(int)}
+```
+
+Record the data-generating parameters
+
+```{python}
+true_params = {
+    'beta_0': sim_data.beta_0[0],
+    'pct_sex': sim_data.pct_sex[0],
+    'beta_sex': sim_data.beta_sex[0],
+    'pct_age': sim_data.pct_age[0],
+    'beta_age':sim_data.beta_age[0],
+    'pct_eth': sim_data.pct_eth[0],
+    'beta_eth':sim_data.beta_eth[0],
+    'pct_edu': sim_data.pct_edu[0],
+    'beta_edu':sim_data.beta_edu[0]
+}
+true_params
+```
+
+#### Model 1: `sum_to_zero_vector`
+
+This model is in file [binomial_4_preds_ozs.stan](https://github.com/stan-dev/example-models/tree/master/jupyter/sum-to-zero/stan/binomial_4_preds_ozs.stan).
+
```

Comment:
I suggest showing at least the parameters block for the `sum_to_zero_vector`. For the below python code, I suggest putting all the following code in one block.